cishouseholds/derive.py

```diff
@@ -16,6 +19,29 @@
 from cishouseholds.pyspark_utils import get_or_create_spark_session
 
 
+def assign_column_value_from_multiple_column_map(
+    df: DataFrame, column_name_to_assign: str, column_name_to_map: List[List[Any]], column_names: List[str]
+):
+    """
+    assign column value based on values of any number of columns in a dictionary
```

Comment:
I see how it works, but 'm struggling to wrap my head around the inputs to this function. Please could you expand the docstring to describe how `column_name_to_map` is structured and what `column_names` are?